### PR TITLE
don't return errors, throw them

### PR DESCRIPTION
**Pull request by TimeForANinja**
rather throw them
also adjust README & docs

follows the convention but will propably break some projects

BREAKING CHANGES:
- the following helper functions can now throw an error instead of returning an error: `ytdl.chooseFormat`, `ytdl.getURLVideoID`, `ytdl.getVideoID`

fixes https://github.com/fent/node-ytdl-core/issues/485

### CODE REVIEW

**Comment by TimeForANinja** [comment]
🤔 guess i should specify the throw-reason in tests

**Comment by fent** [comment]
looks like `throws Error` is not valid typescript syntax. but you can use `| never`

https://stackoverflow.com/a/49435648/157629

**Comment by Sofian93** [comment]
I have the error Can not find module 'm3u8stream / lib / parse-time' but I do not know where it comes from (because I have not defined this module) and I do not know how to fix it. Thanks to the one who will help me!

**Comment by TimeForANinja** [comment]
@Sofian93 pls don't just comment in some unrelated issue/pull
To solve your Problem: update your dependencies, should be fixed

**Comment by fent** [comment]
ok so this looks good. but since this has breaking changes, I'm going to wait on the other two in-progress PRs, (https://github.com/fent/node-ytdl-core/pull/503 and https://github.com/fent/node-ytdl-core/pull/526) since those are bug fixes.

File: lib/index.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const util        = require('./util');
 const sig         = require('./sig');
 const request     = require('miniget');
 const m3u8stream  = require('m3u8stream');
-const parseTime   = require('m3u8stream/lib/parse-time');
+const parseTime   = require('m3u8stream/dist/parse-time');
```
Comment (anderamko) [inline lib/index.js]: +1 on the solution

Comment (aurelhann) [reply lib/index.js]: yep it's working with it! Merge it please ! :pray:

File: lib/info-extras.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ const qs          = require('querystring');
 const url         = require('url');
 const Entities    = require('html-entities').AllHtmlEntities;
 const util        = require('./util');
-const parseTime   = require('m3u8stream/lib/parse-time');
+const parseTime   = require('m3u8stream/dist/parse-time');
```
Comment (aurelhann) [inline lib/info-extras.js]: for my use with https://www.npmjs.com/package/spotify-dl
i need this too

**Comment by TimeForANinja** [comment]
resolve conflicts 😉

**Comment by SwapnilSoni1999** [comment]
why this aint implemented :(

**Comment by TimeForANinja** [comment]
nice 👍

**Pull request merged**